code/game/objects/items/robot/items/tools.dm

```diff
@@ -174,5 +176,199 @@
 	projectile.speed *= (1 / projectile_speed_coefficient)
 	projectile.cut_overlay(projectile_effect)
 
+//bare minimum omni-toolset for modularity
+/obj/item/borg/cyborg_omnitool
+	name = "cyborg omni-toolset"
+	desc = "You shouldn't see this in-game normally."
+	icon = 'icons/mob/silicon/robot_items.dmi'
+	icon_state = "toolkit_medborg"
+	///our tools
+	var/list/radial_menu_options = list()
+	///The toolspeed for our tools
+	toolspeed = 1
+	var/obj/item/reference
+	//is the toolset upgraded or not
+	var/upgraded = FALSE
+	///how much faster should the toolspeed be?
+	var/upgraded_toolspeed = 0.7
+
+/obj/item/borg/cyborg_omnitool/get_all_tool_behaviours()
+	return list(TOOL_SCALPEL)
+
+/obj/item/borg/cyborg_omnitool/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/butchering, \
+	speed = 8 SECONDS, \
+	effectiveness = 100, \
+	disabled = TRUE, \
+	)
+	radial_menu_options = list(
+		NO_TOOL = image(icon = 'icons/mob/silicon/robot_items.dmi', icon_state = initial(icon_state)),
+		TOOL_SCALPEL = image(icon = 'icons/obj/medical/surgery_tools.dmi', icon_state = "[TOOL_SCALPEL]"),
+	)
+
+/obj/item/borg/cyborg_omnitool/attack_self(mob/user)
+	var/new_tool_behaviour = show_radial_menu(user, src, radial_menu_options, require_near = TRUE, tooltips = TRUE)
+
+	if(isnull(new_tool_behaviour) || new_tool_behaviour == tool_behaviour)
+		return
+	if(new_tool_behaviour == NO_TOOL)
+		tool_behaviour = null
+	else
+		tool_behaviour = new_tool_behaviour
+
+	reference_item_for_parameters()
+	update_tool_parameters(reference)
+	update_appearance(UPDATE_ICON_STATE)
+	playsound(src, 'sound/items/change_jaws.ogg', 50, TRUE)
+
+/// Used to get reference item for the tools
+/obj/item/borg/cyborg_omnitool/proc/reference_item_for_parameters()
+	switch(tool_behaviour)
+		if(TOOL_SCALPEL)
+			reference = /obj/item/scalpel
```

Comment:
are you expecting more?

Reply:
ah, it's just as an example item on how to add new items. should probably add another tool beside that

Reply:
should add `SHOULD_CALL_PARENT(FALSE)` if youre just using it as a reference

Reply:
> should add `SHOULD_CALL_PARENT(FALSE)` if youre just using it as a reference

gotcha, where should I add `SHOULD_CALL_PARENT(FALSE)`? i'm not too familiar with spacemanDMM

Reply:
sorry I shouldve been more specific, it goes here in the parent, ideally top of the proc. this prevents other people from returning ..() etc, which looks to not be what you want here

Reply:
ah, that's what that does. I get it now. thanks for describing it! it's added now